### DISCUSSION TITLE
关于数据sql排序查询相关的疑问，对飞索引的字段排序，往往会造成be down掉

### DISCUSSION THREAD

**Discussion by panys731224085** [apache/doris#29307] [A - General / Q&A]
对一张5000万的表进行数据查询，里面涉及单个字段的筛选，以及其他另一个字段的排序，取前1000，会导致数据be服务直接down掉，这个是我使用的上的问题吗？系统架构是一个1fe 1be的配置。
doris-be-1  | I1229 08:28:47.766223   599 daemon.cpp:211] OS physical memory 125.40 GB. Process memory usage 2.49 GB, limit 119.00 GB, soft limit 107.10 GB. Sys available memory 82.35 GB, low water mark 1.60 GB, warning water mark 3.20 GB. Refresh interval memory growth 0 B
doris-be-1  | I1229 08:28:49.345274   915 exec_node.cpp:196] fragment_instance_id=a87ecd7b2f114f24-a0f1abaa26efb4ec closed
doris-be-1  | I1229 08:28:49.346222   915 exec_node.cpp:196] fragment_instance_id=a87ecd7b2f114f24-a0f1abaa26efb4ec closed
doris-be-1  | I1229 08:28:49.346277   915 plan_fragment_executor.cpp:527] Close() fragment_instance_id=a87ecd7b2f114f24-a0f1abaa26efb4ec
doris-be-1  | I1229 08:28:49.363956  1410 internal_service.cpp:1420] multiget_data single_row, cost(us):15774
doris-be-1  | I1229 08:28:49.671316  1410 internal_service.cpp:1420] multiget_data single_row, cost(us):2459
doris-be-1  | I1229 08:28:49.676146   599 daemon.cpp:211] OS physical memory 125.40 GB. Process memory usage 2.61 GB, limit 119.00 GB, soft limit 107.10 GB. Sys available memory 82.20 GB, low water mark 1.60 GB, warning water mark 3.20 GB. Refresh interval memory growth 0 B
doris-be-1  | I1229 08:28:49.953414  1410 internal_service.cpp:1420] multiget_data single_row, cost(us):2229
doris-be-1  | I1229 08:28:50.079747   599 daemon.cpp:211] OS physical memory 125.40 GB. Process memory usage 2.71 GB, limit 119.00 GB, soft limit 107.10 GB. Sys available memory 82.10 GB, low water mark 1.60 GB, warning water mark 3.20 GB. Refresh interval memory growth 0 B
doris-be-1  | I1229 08:28:50.160282  1410 internal_service.cpp:1420] multiget_data single_row, cost(us):2371
doris-be-1  | I1229 08:28:50.364758  1410 internal_service.cpp:1420] multiget_data single_row, cost(us):2253
doris-be-1  | I1229 08:28:50.483316   599 daemon.cpp:211] OS physical memory 125.40 GB. Process memory usage 2.81 GB, limit 119.00 GB, soft limit 107.10 GB. Sys available memory 82.00 GB, low water mark 1.60 GB, warning water mark 3.20 GB. Refresh interval memory growth 0 B
doris-be-1  | I1229 08:28:50.543725  1410 internal_service.cpp:1420] multiget_data single_row, cost(us):26
doris-be-1  | I1229 08:28:50.549450  1410 internal_service.cpp:1420] multiget_data single_row, cost(us):20
doris-be-1  | I1229 08:28:50.554828  1410 internal_service.cpp:1420] multiget_data single_row, cost(us):19
doris-be-1  | I1229 08:28:50.573398  1410 internal_service.cpp:1420] multiget_data single_row, cost(us):21
doris-be-1  | I1229 08:28:50.576997  1410 internal_service.cpp:1420] multiget_data single_row, cost(us):19
doris-be-1  | I1229 08:28:50.581918  1410 internal_service.cpp:1483] multiget_data finished, cost(us):1233897
doris-be-1  | I1229 08:28:50.587567   908 exec_node.cpp:196] fragment_instance_id=a87ecd7b2f114f24-a0f1abaa26efb4ed closed
doris-be-1  | I1229 08:28:50.587613   908 plan_fragment_executor.cpp:527] Close() fragment_instance_id=a87ecd7b2f114f24-a0f1abaa26efb4ed
doris-be-1  | I1229 08:28:50.587637   908 query_fragments_ctx.h:60] Deregister query/load memory tracker, queryId=a87ecd7b2f114f24-a0f1abaa26efb4eb, Limit=2.00 GB, CurrUsed=-7.22 GB, PeakUsed=2.09 MB
doris-be-1  | I1229 08:28:54.579576  1546 fragment_mgr.cpp:614] query_id: d17b3db955c842dd-a168af1fe49abd8b coord_addr TNetworkAddress(hostname=192.168.10.108, port=9020) total fragment num on current host: 2
doris-be-1  | I1229 08:28:54.579627  1546 fragment_mgr.cpp:665] Register query/load memory tracker, query/load id: d17b3db955c842dd-a168af1fe49abd8b limit: 2.00 GB
doris-be-1  | I1229 08:28:54.579659  1546 plan_fragment_executor.cpp:89] PlanFragmentExecutor::prepare|query_id=d17b3db955c842dd-a168af1fe49abd8b|instance_id=d17b3db955c842dd-a168af1fe49abd8d|backend_num=0|pthread_id=139843941369600
doris-be-1  | I1229 08:28:54.579876  1546 plan_fragment_executor.cpp:89] PlanFragmentExecutor::prepare|query_id=d17b3db955c842dd-a168af1fe49abd8b|instance_id=d17b3db955c842dd-a168af1fe49abd8c|backend_num=1|pthread_id=139843941369600
doris-be-1  | I1229 08:28:54.579928   916 fragment_mgr.cpp:492] PlanFragmentExecutor::_exec_actual|query_id=d17b3db955c842dd-a168af1fe49abd8b|instance_id=d17b3db955c842dd-a168af1fe49abd8d|pthread_id=139848979482368
doris-be-1  | I1229 08:28:54.580564   906 fragment_mgr.cpp:492] PlanFragmentExecutor::_exec_actual|query_id=d17b3db955c842dd-a168af1fe49abd8b|instance_id=d17b3db955c842dd-a168af1fe49abd8c|pthread_id=139849063409408
doris-be-1  | I1229 08:28:54.582429   906 plan_fragment_executor.cpp:225] PlanFragmentExecutor::open|query_id=d17b3db955c842dd-a168af1fe49abd8b|instance_id=d17b3db955c842dd-a168af1fe49abd8c|mem_limit=2.00 GB
doris-be-1  | I1229 08:28:54.582439   916 plan_fragment_executor.cpp:225] PlanFragmentExecutor::open|query_id=d17b3db955c842dd-a168af1fe49abd8b|instance_id=d17b3db955c842dd-a168af1fe49abd8d|mem_limit=2.00 GB
doris-be-1  | I1229 08:28:54.669001   906 exec_node.cpp:196] fragment_instance_id=d17b3db955c842dd-a168af1fe49abd8c closed
doris-be-1  | I1229 08:28:54.669994   906 exec_node.cpp:196] fragment_instance_id=d17b3db955c842dd-a168af1fe49abd8c closed
doris-be-1  | I1229 08:28:54.670032   906 plan_fragment_executor.cpp:527] Close() fragment_instance_id=d17b3db955c842dd-a168af1fe49abd8c
doris-be-1  | F1229 08:28:54.672708  1414 column.h:249] Method insert_many_dict_data is not supported for Int64
doris-be-1  | *** Check failure stack trace: ***
doris-be-1  |     @     0x5576f88c52cd  google::LogMessage::Fail()
doris-be-1  |     @     0x5576f88c7809  google::LogMessage::SendToLog()
doris-be-1  |     @     0x5576f88c4e36  google::LogMessage::Flush()
doris-be-1  |     @     0x5576f88c7e79  google::LogMessageFatal::~LogMessageFatal()
doris-be-1  |     @     0x5576f1257127  doris::vectorized::IColumn::insert_many_dict_data()
doris-be-1  |     @     0x5576f19deca3  doris::segment_v2::BinaryDictPageDecoder::read_by_rowids()
doris-be-1  |     @     0x5576f1997bb6  doris::segment_v2::FileColumnIterator::read_by_rowids()
doris-be-1  |     @     0x5576f1d4b8d8  doris::PInternalServiceImpl::_multi_get()
doris-be-1  |     @     0x5576f1d4c14f  _ZNSt17_Function_handlerIFvvEZN5doris20PInternalServiceImpl13multiget_dataEPN6google8protobuf13RpcControllerEPKNS1_16PMultiGetRequestEPNS1_17PMultiGetResponseEPNS4_7ClosureEEUlvE_E9_M_invokeERKSt9_Any_data
doris-be-1  |     @     0x5576f11be180  doris::PriorityThreadPool::work_thread()
doris-be-1  |     @     0x5576fb111030  execute_native_thread_routine
doris-be-1  |     @     0x7f31ee4b2ea7  start_thread
doris-be-1  |     @     0x7f31ee29edef  clone
doris-be-1  |     @              (nil)  (unknown)
doris-be-1  | *** Query id: 0-0 ***
doris-be-1  | *** Aborted at 1703838534 (unix time) try "date -d @1703838534" if you are using GNU date ***
doris-be-1  | *** Current BE git commitID: Unknown ***
doris-be-1  | *** SIGABRT unknown detail explain (@0x208) received by PID 520 (TID 1414 OR 0x7f30367b4700) from PID 520; stack trace: ***
doris-be-1  | I1229 08:28:55.102461   599 daemon.cpp:211] OS physical memory 125.40 GB. Process memory usage 2.95 GB, limit 119.00 GB, soft limit 107.10 GB. Sys available memory 81.88 GB, low water mark 1.60 GB, warning water mark 3.20 GB. Refresh interval memory growth 0 B
doris-be-1  | I1229 08:28:55.404728   599 daemon.cpp:211] OS physical memory 125.40 GB. Process memory usage 3.06 GB, limit 119.00 GB, soft limit 107.10 GB. Sys available memory 81.86 GB, low water mark 1.60 GB, warning water mark 3.20 GB. Refresh interval memory growth 0 B
doris-be-1  |  0# doris::signal::(anonymous namespace)::FailureSignalHandler(int, siginfo_t*, void*) at /root/doris/be/src/common/signal_handler.h:413
doris-be-1  |  1# 0x00007F31EE4BE140 in /lib/x86_64-linux-gnu/libpthread.so.0
doris-be-1  |  2# raise in /lib/x86_64-linux-gnu/libc.so.6
doris-be-1  |  3# abort in /lib/x86_64-linux-gnu/libc.so.6
doris-be-1  |  4# 0x00005576F88CFCB9 in /opt/apache-doris/be/lib/doris_be
doris-be-1  |  5# 0x00005576F88C52CD in /opt/apache-doris/be/lib/doris_be
doris-be-1  |  6# google::LogMessage::SendToLog() in /opt/apache-doris/be/lib/doris_be
doris-be-1  |  7# google::LogMessage::Flush() in /opt/apache-doris/be/lib/doris_be
doris-be-1  |  8# google::LogMessageFatal::~LogMessageFatal() in /opt/apache-doris/be/lib/doris_be
doris-be-1  |  9# doris::vectorized::IColumn::insert_many_dict_data(int const*, unsigned long, doris::StringRef const*, unsigned long, unsigned int) at /root/doris/be/src/vec/columns/column.h:249
doris-be-1  | 10# doris::segment_v2::BinaryDictPageDecoder::read_by_rowids(unsigned int const*, unsigned long, unsigned long*, COW<doris::vectorized::IColumn>::mutable_ptr<doris::vectorized::IColumn>&) at /root/doris/be/src/olap/rowset/segment_v2/binary_dict_page.cpp:297
doris-be-1  | 11# doris::segment_v2::FileColumnIterator::read_by_rowids(unsigned int const*, unsigned long, COW<doris::vectorized::IColumn>::mutable_ptr<doris::vectorized::IColumn>&) at /root/doris/be/src/olap/rowset/segment_v2/column_reader.cpp:1063
doris-be-1  | 12# doris::PInternalServiceImpl::_multi_get(doris::PMultiGetRequest const*, doris::PMultiGetResponse*) at /root/doris/be/src/service/internal_service.cpp:1441
doris-be-1  | 13# std::_Function_handler<void (), doris::PInternalServiceImpl::multiget_data(google::protobuf::RpcController*, doris::PMultiGetRequest const*, doris::PMultiGetResponse*, google::protobuf::Closure*)::{lambda()#1}>::_M_invoke(std::_Any_data const&) at /var/local/ldb-toolchain/include/c++/11/bits/std_function.h:291
doris-be-1  | 14# doris::PriorityThreadPool::work_thread(int) at /root/doris/be/src/util/priority_thread_pool.hpp:147
doris-be-1  | 15# execute_native_thread_routine in /opt/apache-doris/be/lib/doris_be
doris-be-1  | 16# start_thread in /lib/x86_64-linux-gnu/libpthread.so.0
doris-be-1  | 17# __clone in /lib/x86_64-linux-gnu/libc.so.6

sql：SELECT * FROM `table_name` WHERE `from_address` = '###################' order by `to_address` LIMIT 0,1000
1105 - RpcException, msg: org.apache.doris.rpc.RpcException: io.grpc.StatusRuntimeException: UNAVAILABLE: Network closed for unknown reason

**Comment by LemonLiTree** [comment]
which version？

**Reply by panys731224085** [reply]
docker     apache/doris:2.0.0_alpha-fe-x86_64

**Reply by panys731224085** [reply]
be version apache/doris:2.0.0_alpha-be-x86_64

**Reply by LemonLiTree** [reply]
方便的话升级看看？这个版本 太旧了
